bookkeeper-server/src/main/java/org/apache/bookkeeper/client/BookKeeperAdmin.java

```diff
@@ -798,13 +835,20 @@ public void openComplete(int rc, final LedgerHandle lh, Object ctx) {
                         @Override
                         public void openComplete(int newrc, final LedgerHandle newlh, Object newctx) {
                             if (newrc != BKException.Code.OK) {
-                                LOG.error("BK error close ledger: " + lId, BKException.create(newrc));
-                                finalLedgerIterCb.processResult(newrc, null, null);
+                                if (skipUnrecoverableLedgers) {
```

Comment:
Do we need to check the rc is `NoSuchEntryException` ?

Reply:
I think we don't need to check `NoSuchEntryException`. We are just opening a ledger not read data and it looks like there is no way to throw that exception?

Reply:
For asyncOpenLedger operation, the rc only has `BKException.Code.ReadException` or `Exception.Code.OK`
```java
lh.asyncReadLastConfirmed(new ReadLastConfirmedCallback() {
                @Override
                public void readLastConfirmedComplete(int rc,
                        long lastConfirmed, Object ctx) {
                    if (rc != BKException.Code.OK) {
                        closeLedgerHandle();
                        openComplete(bk.getReturnRc(BKException.Code.ReadException), null);
                    } else {
                        lh.lastAddConfirmed = lh.lastAddPushed = lastConfirmed;
                        openComplete(BKException.Code.OK, lh);
                    }
                }
            }, null);
```
So we doesn't need to check `NoSuchEntryException`.